manuscript.md

```diff
@@ -124,12 +124,14 @@ Apart from the engagement needed from the community to foster the development of
 
 # The community can contribute to development
 
-Open-source software also plays a critical role in iterative refinement.
-Bugs and mistakes are inevitable in complex software, and these issues are often best addressed through collaborative scrutiny.
-In our own work, users have uncovered bugs that we subsequently corrected, improving the reliability and robustness of our tools.
-We have also experienced users asking questions about the underlying code, leading to new features and more efficient algorithms.
-This feedback loop is unique to OSS, where the contributions from the community enhance the quality and precision of the software over time.  Compared to proprietary software, OSS can often move faster and defray development costs by enabling users to build and contribute the features they need, rather than hoping that the maintainers of the software are willing or able to add the features themselves.
-Without such transparency, computational research can become a "black box" that stifles innovation rather than promoting it, hindering the growth of scientific knowledge.
+One of the greatest strengths of open-source software is that "given enough eyeballs, all bugs are shallow" (http://www.catb.org/~esr/writings/cathedral-bazaar/cathedral-bazaar/ar01s04.html). 
```

Comment:
That is a very good point. I just feel it does not belong into this paper. We are talking here about Open Source Science Software, and more specifically about Proteomics software. I doubt that the majority of software has/had enough eyeballs and reviews - so making this point could fire back. As consequence the security level of many smaller projects is also not that great. E.g. pytorch is IMHO still consuming arbitrary code (in form of models)?

Reply:
I will vote for keeping that idea. While it can be fireback, it is mostly the case that OSS enables the quick detection of bugs and improves the code. While proteomics is poorly understood because the community is small and few people review code and do bug fixing, it is the case for mainly any OSS software. I have to say that in the case of some packages like TRFP, we do have a lot of contributors fixing and detecting bugs. But it could be a unique or rare example.

Reply:
I don't think it means that all OSS **has** been examined, but that it **can** be. The manuscript goes on to highlight how "In our own work, users have uncovered bugs that we subsequently corrected..."

Is PyTorch a "small" project? I certainly wouldn't say so. Nor would I say (since we cannot know in the absence of source code) that commercial projects are any more secure. Even large projects have security flaws all the time, particularly when written in non-memory safe languages. Most of these kinds of flaws are far beyond the skill level of scientists-turned-programmers to detect, let alone even be cognizant of.

Reply:
I know and in theory, this might be true. In practice, it's not. Even projects like openssh have been badly maintained. And science has really no reputation for developing safe software. We send python pickles or R sessions/objects around and do not care about security. This security aspect is always put forward for OSS and its debunked very quickly because theory and reality do not match at all. We have other strong arguments and I think we don't need to this one - which is in practise wrong.

Reply:
Where is there any argument being made about security in the manuscript or the highlighted section? Bugs != security.

Reply:
> Bugs != security.

Would you agree that it is a subset? And all arguments are true for non-security bugs as well?
Adding `"given enough eyeballs, all bugs are shallow"` to the paper but knowing that we don't have those eyeballs, in proteomics, not in science and maybe not even bigger open source projects just feels wrong to me. But feel free to add it.

Reply:
It's definitely a subset, but I think in the context of proteomics a very small subset relative to logic bugs. Most programs are running on trusted inputs (data from a user's instruments, etc). But I am also OK with removing the statement if you feel strongly about it.

Reply:
I'm in favor of keeping it, mostly because I've been the recipient of folks catching my own bugs 🫣

Reply:
As a side note, when I review papers, I have a policy of not downloading and running the authors' closed-source tools mostly as a security precaution. For open-source tools, I look through the source code then choose to run them or not.

Obviously I do use closed-source tools that I have to trust for other reasons - this is merely for peer review. I don't assert that this is right or wrong, just a personal policy.

Reply:
I'm in the middle here. I think we should keep it, but not focus on security but on other aspects of the software bugs. We can also point it out that while in principle the idea of OSS can benefit bug fixing etc, it will only be successful if the community and contributors review the code, and participate in the project.